### ISSUE TITLE
Command line script execution takes aaaaaaaages

### ISSUE THREAD

**Issue by o-o00o-o** [otykier/TabularEditor#324]
Hi Daniel, we have noticed that our build process was taking a really long time and the main culprit was a Tabular Editor script task. This script runs in around 1 second in the UI but takes over 5 minutes in command line

Nothing obvious using msbuild but when running the command below manually 

start /b /wait 'C:\Users\bgerhardi\Source\Vega\VegaDW\build\TabularEditor.exe' C:\Users\bgerhardi\Source\Vega\VegaDW\src\VegaDW.Cube\database.json -SCRIPT "C:\Users\bgerhardi\Source\Vega\VegaDW\build\RecreateAllAutoDerivatives.cs" -BIM  C:\Users\bgerhardi\Source\Vega\VegaDW\output\Model.bim -OVERWRITE -VSTS

I get hundreds of 
Dependency tree built in 214 ms
Dependency tree built in 212 ms
Dependency tree built in 207 ms
Dependency tree built in 278 ms

scrolling through. I guess it is the writing of these messages that is causing the long delay.

We are running v2.7.4 still but is there any way that we can stop these messages and have this run faster?

**Comment by otykier** [comment]
Thanks for reporting. I'll provide a fix for the latest version (2.8.5).

To work around the issue with any other version of Tabular Editor, surround your script with the following code:

```csharp
dynamic handler = Model.GetType().GetFields(
    System.Reflection.BindingFlags.Instance |
    System.Reflection.BindingFlags.NonPublic
).FirstOrDefault(f => f.Name == "Handler").GetValue(Model);

handler.BeginUpdate("script");

// ...
// < YOUR SCRIPT HERE >
// ...

handler.EndUpdate();
```

**Comment by o-o00o-o** [comment]
Thanks Daniel, I'm getting the following error

```
##vso[task.logissue type=error;sourcepath=C:\Users\bgerhardi\Source\Vega\VegaDW\build\RecreateAllAutoDerivatives.cs;l
  inenumber=0;columnnumber=0;code=CS1701]Assuming assembly reference 'Microsoft.AnalysisServices.Tabular, Version=15.7.
  0.2, Culture=neutral, PublicKeyToken=89845dcd8080cc91' matches 'Microsoft.AnalysisServices.Tabular, Version=18.0.5.0,
   Culture=neutral, PublicKeyToken=89845dcd8080cc91', you may need to supply runtime policy
  ##vso[task.logissue type=error;sourcepath=C:\Users\bgerhardi\Source\Vega\VegaDW\build\RecreateAllAutoDerivatives.cs;l
  inenumber=0;columnnumber=0;code=CS0518]Predefined type 'Microsoft.CSharp.RuntimeBinder.Binder' is not defined or impo
  rted
  ##vso[task.logissue type=error;sourcepath=C:\Users\bgerhardi\Source\Vega\VegaDW\build\RecreateAllAutoDerivatives.cs;l
  inenumber=42;columnnumber=1;code=CS1969]One or more types required to compile a dynamic expression cannot be found. A
  re you missing a reference?
  ##vso[task.logissue type=error;sourcepath=C:\Users\bgerhardi\Source\Vega\VegaDW\build\RecreateAllAutoDerivatives.cs;l
  inenumber=322;columnnumber=1;code=CS1969]One or more types required to compile a dynamic expression cannot be found.
  Are you missing a reference?
```

Any ideas?

**Comment by o-o00o-o** [comment]
also that reminds me, if I try to run this in the UI I get this

![image](https://user-images.githubusercontent.com/13693159/64981504-1801e100-d8b4-11e9-927f-928d33efba3b.png)

It seems to give me this now any time I have any syntax errors in my script (instead of actually giving me the errors)

We install (unzip) TabularEditor to our build folder so all the libraries are in the same folder. You can see the versions:

![image](https://user-images.githubusercontent.com/13693159/64981692-7e86ff00-d8b4-11e9-8fe2-c2c140361325.png)

any ideas on how to resolve this also? Perhaps we just need to upgrade.

**Comment by otykier** [comment]
The errors you are seeing indicate a mismatch between the version of the Microsoft.AnalysisServices DLL's available, and the ones used by the version of Tabular Editor that you're running.

To make sure that you have the correct versions, grab the .MSI file of 2.7.4, install it locally on your machine, and then copy the DLL's from the installation folder to your build folder. Alternatively, I think it should be possible to download older versions of Microsoft.AnalysisServices DLLs from NuGet: https://www.nuget.org/api/v2/package/Microsoft.AnalysisServices.retail.amd64/15.7.0.2 - if you unzip the .nupkg file, it should contain the DLLs as well.

**Comment by otykier** [comment]
If you're still having issues with 2.8.5 using the latest version of the Microsoft.AnalysisServices DLLs, try to delete the TOMWrapper.dll / TOMWrapper14.dll file from your `%AppData%\..\Local\TabularEditor` folder and restart Tabular Editor.

**Comment by o-o00o-o** [comment]
Thanks for the suggestions Daniel, it helped me work out what was going on, 

it turned out that another part of our build setup process was overwriting the dll's with a different version. I've isolated it now in its own folder to prevent this.

**Comment by otykier** [comment]
Fixed in [2.8.6](https://github.com/otykier/TabularEditor/releases/tag/2.8.6).
